contracts/internals/ensResolvable.sol

```diff
@@ -28,9 +28,10 @@ contract ENSResolvable {
     /// @notice _ensRegistry points to the ENS registry smart contract.
     address private _ensRegistry;
 
-    /// @param _ensReg_ is the ENS registry used
-    constructor(address _ensReg_) internal {
-        _ensRegistry = _ensReg_;
+    /// @notice Checks if the contract has been initialized succesfully i.e. the ENS registry has been set.
+    modifier initialized() {
```

Comment:
I think we should keep the constructors or do what I did in the controller v2 branch which is to remove the constructors and set a default value for _ensRegistry

Reply:
All these comments are good point and as I wrote before I found that there is a bit of inconsistency on the way we initialize things.

Reply:
The constructors for inherited contracts can indeed be kept (even though the guideline is to avoid them for upgradeable contracts) It is just code replication and in the end, you always have to invoke them (in many cases we forgot to invoke them and our tests were failing and it was hard to debug them). If it is just for aesthetic reasons, my guess is that we can keep them.

Reply:
I have looked at these too. I am indifferent to changing them now or in the future, but :

```
I like the default value set in the Controllable.sol :

_initializeControllable


This one is different and I prefer the above way of doing things : 

_initializeENSResolvable```

Reply:
by the way, if we follow this approach then we can even remove the harcoded ens values in the deployer contract. but every time we have to edit separate contracts. I ' fine with that, but let's change in a separate PR and think a bit of what we want.

Reply:
Yeah, I've read the openzeppelin docs and I'm OK with the initializer instead of constructor for wallet contracts.